migration/incremental/032-logical-epoch-to-time-epoch.sql

```diff
@@ -0,0 +1,96 @@
+CREATE TABLE _prom_catalog.global_epoch (
+    current_epoch TIMESTAMPTZ NOT NULL,
+    delete_epoch TIMESTAMPTZ NOT NULL,
+    -- force there to only be a single row
+    is_unique BOOLEAN NOT NULL DEFAULT true CHECK (is_unique = true),
+    UNIQUE (is_unique)
+);
```

Comment:
You can force a single row without the extra column like this:

CREATE TABLE _prom_catalog.global_epoch (
    current_epoch TIMESTAMPTZ NOT NULL,
    delete_epoch TIMESTAMPTZ NOT NULL
);
INSERT INTO _prom_catalog.global_epoch (current_epoch, delete_epoch) values ('-infinity', '-infinity');
CREATE UNIQUE INDEX global_epoch_unique_idx ON _prom_catalog.global_epoch ((true));
You have to insert the one row before creating the unique index, but I see that as a bonus because your code can assume the row will always already exist.

However, I'm not sure if/how it would work with restores, so maybe what you have is best.

Reply:
I have adopted this suggestion as well. Let's see if the build likes it.